File: frontend/ckd-care-app/src/api/slump.ts
import { api } from "./client";

export type MicroChallengeCode =
  | "HYDRATION_CUP"
  | "EXERCISE_STRETCH"
  | "DIET_VEGGIE"
  | "SLEEP_EARLY"
  | "STRESS_BREATH";

export interface MicroChallenge {
  code: MicroChallengeCode;
  category: string;
  title: string;
  icon: string;
  minutes: number;
  hint: string;
}

export interface SlumpStatusResponse {
  is_slump: boolean;
  days_since_last_checkin: number;
  threshold_days: number;
  micro: MicroChallenge;
  already_checked_in_today: boolean;
}

export interface SlumpMicroCheckinResponse {
  recovered: boolean;
  micro_code: MicroChallengeCode;
  checked_at: string;
  message: string;
}

export const slumpApi = {
  status: () => api.get<SlumpStatusResponse>("/challenges/slump-micro"),
  checkin: (micro_code: MicroChallengeCode) =>
    api.post<SlumpMicroCheckinResponse>("/challenges/slump-micro/checkin", { micro_code }),
};

File: app/dtos/slump.py
from datetime import datetime
from typing import Annotated

from pydantic import BaseModel, Field

from app.models.slump import MicroChallengeCode


class MicroChallengeDTO(BaseModel):
    code: MicroChallengeCode
    category: str
    title: str
    icon: str
    minutes: int
    hint: str


class SlumpStatusResponse(BaseModel):
    is_slump: bool
    days_since_last_checkin: int
    threshold_days: int
    micro: MicroChallengeDTO
    already_checked_in_today: bool


class SlumpMicroCheckinRequest(BaseModel):
    micro_code: Annotated[MicroChallengeCode, Field(description="체크인할 마이크로 챌린지 코드")]


class SlumpMicroCheckinResponse(BaseModel):
    recovered: bool
    micro_code: MicroChallengeCode
    checked_at: datetime
    message: str
